### PR TITLE
Separate bgio server and FBG

### PR DESCRIPTION
**Pull request by jasonharrison**
Separate bgio server and FBG (allows for updates of FBG without having to stop the bgio server)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,5 +23,6 @@ RUN yarn install
 RUN yarn run hexo generate
 
 WORKDIR /appdata
+
 # start server
```
Comment (flamecoals) [inline Dockerfile]: Why do you have two docker files?

File: package.json [hunk 2]
```diff
@@ -15,7 +15,9 @@
     "prod:build": "run-s server:build webpack:build",
     "precommit": "yarn run verify-lint && yarn run verify-tsfmt && yarn run server:build && yarn run test",
     "pre": "yarn run precommit",
-    "server": "node server-build/server.js",
+    "server:bgio": "node server-build/server_bgio.js",
```
Comment (flamecoals) [inline package.json]: I think you can leave "server" executing both

File: src/App/Game/Game.tsx [hunk 3]
```diff
@@ -123,7 +123,9 @@ export default class Game extends React.Component<IGameProps, {}> {
         clientConfig.ai = state.ai.bgioAI(aiLevel);
       }
       if (this.mode === GameMode.OnlineFriend) {
-        clientConfig.multiplayer = true;
+        const server = typeof window !== 'undefined' ?
+          process.env.BGIO_SERVER_URL || `${window.location.hostname}:8001` : 'localhost:8001';
```
Comment (flamecoals) [inline src/App/Game/Game.tsx]: When would the window be undefined?

File: webpack.dev.js [hunk 4]
```diff
@@ -12,10 +13,13 @@ module.exports = merge(common, {
   plugins: [
     new WebpackShellPlugin({
        onBuildEnd: {
-         scripts: ['node server-build/server.js'],
+         scripts: ['node server-build/server_fbg.js','node server-build/server_bgio.js'],
```
Comment (flamecoals) [inline webpack.dev.js]: Does both execute in parallel in this case?

File: Dockerfile (hunk 1 above)
Comment (jasonharrison) [reply Dockerfile]: Left over by accident, removed bgio.Dockerfile

File: src/App/Game/Game.tsx (hunk 3 above)
Comment (jasonharrison) [reply src/App/Game/Game.tsx]: Hrmph.  I think I ran into an issue testing with Jest when I added this, but I don't remember/can't reproduce now, so removing this.

File: webpack.dev.js (hunk 4 above)
Comment (jasonharrison) [reply webpack.dev.js]: Yes :)

File: src/App/Game/Game.tsx (hunk 3 above)
Comment (jasonharrison) [reply src/App/Game/Game.tsx]: Aha.  It was for SSR.

```
ReferenceError: window is not defined
      at Game.render (webpack:///./src/App/Game/Game.tsx?:132:58)
      at /appdata/node_modules/react-tree-walker/dist/react-tree-walker.js:211:33
      at /appdata/node_modules/react-tree-walker/dist/react-tree-walker.js:141:36
```

**Pull request merged**